tests/unit/test_ingress_per_unit_scenario.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for the ingress-per-unit mode of haproxy charm."""
```

Comment:
Can you rename this test file to `test_charm.py`. This actually tests the charm.py file, so let's follow conventions. Also, there's no need to suffix the tests with `_scenario`